examples/dart/test/sync_multiple_processes_test.dart

```diff
@@ -60,4 +68,32 @@ void main() {
       // since both realm connections are only for 1 realm, this deletes both
     });
   });
+  // Note: This test doesn't actually test what is being documented, b/c the
+  // documentation is about working across multiple processes and it's not
+  // intuitive to do that within a unit test (which runs in only 1 process)
+  test("Realm.refresh()", () async {
+    final realm = Realm(Configuration.local([Person.schema]));
+    realm.write(() => realm.add(Person('John')));
+    // :snippet-start: refresh-main-process
+    // Delete object in one process
+    realm.write(() {
+      realm.add(Person('John'));
+    });
+    // :snippet-end:
+    // :snippet-start: refresh-secondary-process
+    // You must call realm.refresh() in the secondary process
+    // before the data written in the main process registers
+    // in the secondary process.
+    realm.refresh();
+    realm.find<Person>('John');
+    // :snippet-end:
+    // :snippet-start: refresh-async
+    // Asynchronously refresh the realm in the background.
+    realm.refreshAsync();
```

Comment:
```suggestion
    realm.refreshAsync();
```
This is a bad practice. One should really use await with async calls.

Reply:
when you call a method with await you release the main isolate (thread/event loop) to handle any other async events and microtasks. Then when the result is available the code continues with the next statement.

Reply:
thx, taking conversation to slack for additional clarification.